.gitlab/build-oci.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+source common_build_functions.sh
+
+if [ -n "$CI_COMMIT_TAG" ] && [ -z "$DOTNET_PACKAGE_VERSION" ]; then
+  DOTNET_PACKAGE_VERSION=${CI_COMMIT_TAG##v}
+fi
+
+if [ -z "$ARCH" ]; then
+  ARCH=amd64
+fi
+
+TMP_DIR=$(mktemp --dir)
+
+curl --location --fail \
+  --output $TMP_DIR/datadog-dotnet-apm.old \
+  "https://github.com/DataDog/dd-trace-dotnet/releases/download/v$DOTNET_PACKAGE_VERSION/datadog-dotnet-apm_${DOTNET_PACKAGE_VERSION}_$ARCH.deb"
+
+fpm --input-type deb \
+  --output-type dir \
+  --name datadog-dotnet-apm \
+  --package $TMP_DIR \
+  $TMP_DIR/datadog-dotnet-apm.old
+
+echo -n $DOTNET_PACKAGE_VERSION > auto_inject-dotnet.version
+
+cp auto_inject-dotnet.version $TMP_DIR/datadog-dotnet-apm.dir/opt/datadog/version
+
+# Build packages
+datadog-package create \
+    --version="$DOTNET_PACKAGE_VERSION" \
+    --package="datadog-apm-library-dotnet" \
+    --archive=true \
+    --archive-path="datadog-apm-library-dotnet-$DOTNET_PACKAGE_VERSION-$ARCH.tar" \
```

Comment:
Do you have any plans to support Windows containers with this? If so, should the package name contain a platform too?

Reply:
A single OCI package contains all targets (arch/os) as different layers, which can be downloaded indivually.
The name of the tar archive in the CI doesn't really matter, because we merge everything into a single OCI package during the release-package step. The archive just needs to contain the arch and OS so we don't overwrite artifacts across jobs.